core/src/main/scala/cats/instances/hash.scala

```diff
@@ -0,0 +1,17 @@
+package cats
+package instances
+
+import cats.functor._
+
+/**
+ * @author Tongfei Chen
+ */
+trait HashInstances {
+  implicit val catsFunctorContravariantForHash: Contravariant[Hash] =
+    new Contravariant[Hash] {
+      /**
+       * Derive an `Hash` for `B` given an `Hash[A]` and a function `B => A`.
+       */
+      def contramap[A, B](ha: Hash[A])(f: B => A): Hash[B] = Hash.by(f)(ha)
```

Comment:
I think this can be also `Cartesian[Hash]`: you can hash each item then combine them in one of many ways. Maybe the fact that there are many ways to combine means it should not be implicit, but we can give functions that do so.

Reply:
Changed to `ContravariantCartesian`. Thanks! I didn't know the existence of `Cartesian` before.